src/main/java/io/litmuschaos/util/HttpResponseHandler.java

```diff
@@ -26,21 +29,18 @@ public <T> T handleResponse(Response response, Class<T> responseType) throws Api
         return transform(response.body().string(), responseType);
     }
 
-    public <T> T handleResponse(Response response, Type responseType) throws ApiException, IOException {
-        if (!response.isSuccessful()) {
-            throw new ApiException(response);
-        }
-        if (response.body() == null) {
-            return transform("", responseType);
+    private <T> T transform(String responseBody, Type responseType) {
+        JsonElement jsonElement = JsonParser.parseString(responseBody);
+
+        if (jsonElement.isJsonObject()) {
+            JsonObject jsonObject = jsonElement.getAsJsonObject();
+            if (jsonObject.has("data")) {
```

Comment:
Good work @somln ! 🚀

However, I believe that the process of checking the `data` field in the `jsonObject` and parsing it is a new feature. Could you separate this functionality into a different PR? That way, we can keep the scope of this one more focused.

Reply:
Got it! I will modify it to remove that logic.